### PR TITLE
build: add esm bundle

### PR DESCRIPTION
**Pull request by nieyuyao**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch.
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue-test-utils/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

Provides an ESM bundle.

### CODE REVIEW

**Review by lmiller1990 — COMMENTED** [review]
Cool, can you also add the `exports` field like so?

https://github.com/vuejs/test-utils/blob/main/package.json#L9-L15

**Review by lmiller1990 — APPROVED** [review]
Thanks, will do a release soon :)

File: packages/test-utils/package.json [hunk 1]
```diff
@@ -3,6 +3,16 @@
...
+  "module": "dist/vue-test-utils.esm.js",
+  "exports": {
+    ".": {
+      "import": {
+        "node": "./dist/vue-test-utils.js",
+        "require": "./dist/vue-test-utils.js",
+        "default": "./dist/vue-test-utils.esm.js"
+      }
+    }
+  },
```
Comment (wolfgangwalther) [inline packages/test-utils/package.json:15]: Why was this added as-is, i.e. why with the `import` nesting? That breaks importing `@vue/test-utils` v1.3.1 in jest (non esm) for me with:

```
Cannot find module '@vue/test-utils' from '...'

    > 1 | import { createLocalVue } from '@vue/test-utils'
```

when I remove the `import` like this:

```json
  "exports": {
    ".": {
      "node": "./dist/vue-test-utils.js",
      "require": "./dist/vue-test-utils.js",
      "default": "./dist/vue-test-utils.esm.js"
    }
  },
```

then it works again.

The comment that suggested to add the `exports` key was referencing this, i.e. the vue3 version of test-utils:

https://github.com/vuejs/test-utils/blob/main/package.json#L9-L15

There is no nested `import` there.

Maybe the following would be best?

```json
  "exports": {
    ".": {
      "import": "./dist/vue-test-utils.esm.js",
      "require": "./dist/vue-test-utils.js",
      "default": "./dist/vue-test-utils.js"
    }
  },
```

That would be the most similar to what is done in vue/test-utils right now.

Comment (lmiller1990) [reply packages/test-utils/package.json:15]: Uh oh, did not realize this broke things. I'll patch this right now.

Comment (lmiller1990) [reply packages/test-utils/package.json:15]: https://github.com/vuejs/vue-test-utils/pull/2020

Comment (lmiller1990) [reply packages/test-utils/package.json:15]: Done
https://github.com/vuejs/vue-test-utils/releases/tag/v1.3.2

Gave it a quick test, seems okay, LMK if you run into any problems.

Comment (wolfgangwalther) [reply packages/test-utils/package.json:15]: Thanks that fixed it!

**Pull request merged**